util/llpcInternal.cpp

```diff
@@ -77,20 +77,27 @@ Function* GetEntryPoint(
 }
 
 // =====================================================================================================================
-// Emits a LLVM function call (inserted before the specified instruction), builds it automically based on return type
-// and its parameters.
-CallInst* EmitCall(
-    StringRef                     funcName,         // Name string of the function
-    Type*                         pRetTy,           // [in] Return type
-    ArrayRef<Value *>             args,             // [in] Parameters
-    ArrayRef<Attribute::AttrKind> attribs,          // Attributes
-    Instruction*                  pInsertPos)       // [in] Where to insert this call
+/// Emits a LLVM function call using the given builder. The callee is built automically based on return
+/// type and its parameters.
+///
+/// @oaram [in] funcName Name of the callee
+/// @param [in] pRetTy Return type of the callee
+/// @param [in] args Arguments to pass to the callee
+/// @param [in] attribs Function attributes
+/// @param [in] builder The IRBuilder to use for creating the call
+llvm::CallInst* EmitCall(
+    StringRef                           funcName,
+    Type*                               pRetTy,
+    ArrayRef<llvm::Value *>             args,
+    ArrayRef<llvm::Attribute::AttrKind> attribs,
+    IRBuilder<>&                        builder)
 {
-    Module* pModule = pInsertPos->getModule();
+    Module* pModule = builder.GetInsertBlock()->getParent()->getParent();
     Function* pFunc = dyn_cast_or_null<Function>(pModule->getFunction(funcName));
-    if (pFunc == nullptr)
+    if (!pFunc)
```

Comment:
pFunc -> pFunc != nullptr

Reply:
I think `!boolean `is acceptable because the meaning is "not this". For a pointer `!ptr`, we have to understand it as "ptr not exist". Under such case, `ptr == nullptr` is very clear in human brain. This was widely discussed within 3D driver team many years ago. For pointer, always explicitly compare with `nullptr`.

Reply:
The 3D driver team's opinion is frankly not relevant when working on a compiler.

Using `!ptr` to check for pointer-nullness has been established canonical practice in C and C++ for **decades**. I'm not going to change this.